processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -44,34 +47,57 @@
     return decorated;
   }
 
+  /**
+   * Returns decorated post-aggregators, based on original un-decorated post-aggregators. In addition, this method
+   * also verifies that there are no output name collisions, and that all of the post-aggregators' required input
+   * fields are present.
+   *
+   * @param otherOutputNames names of fields that will appear in the same output namespace as aggregators and
+   *                         post-aggregators. For most built-in query types, this is either empty, or the list of
+   *                         dimension output names.
+   * @param aggFactories     aggregator factories for this query
+   * @param postAggs         post-aggregators for this query
+   *
+   * @return decorated post-aggregators
+   *
+   * @throws NullPointerException     if otherOutputNames or aggFactories is null
+   * @throws IllegalArgumentException if there are any output name collisions or missing post-aggregator inputs
+   */
   public static List<PostAggregator> prepareAggregations(
+      List<String> otherOutputNames,
```

Comment:
I'm not sure, good question. I guess we might as well consider it public, since it's likely that people will have copied it from builtin query impls.